sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -36,6 +36,18 @@ import org.apache.spark.sql.util.CaseInsensitiveStringMap
 class UnresolvedException(function: String)
   extends AnalysisException(s"Invalid call to $function on unresolved object")
 
+case class PlanWithUnresolvedIdentifier(identifierExpr: Expression,
```

Comment:
In general, I think it's OK to have multiple ways to hold the unresolved identifier. E.g., `UnresolvedRelationIdentifierClause` is a good solution for plans that needs `UnresolvedRelation`, which includes FROM, INSERT, UPDATE and MERGE. We can use `PlanWithUnresolvedIdentifier` for others so that we don't need to create many `XXXIdentifierClause`.

Shall we use the same idea for expressions(attribute and function)?

Reply:
Wouldn't it be nicer to have one way. Are you saying there is no way to handle MERGE using some form of withIdentClause(). What about CREATE TABLE LIKE (that's a DDL command)